internal/lang/evalchecks/eval_for_each_test.go

```diff
@@ -352,12 +361,14 @@ func TestEvaluateForEachExpression_multi_errors(t *testing.T) {
 	}
 }
 
-func TestEvaluateForEachExpressionKnown(t *testing.T) {
+// These test checks if unknown values are valid when being evaluated during the validate phase
```

Comment:
Thank you for this comment!

Reply:
Side question- are they also testing for `plan` phase, or are they only tested during the `validate` phase?

Reply:
Another thought, it would be awesome if you could update the previous test functions with the same quality comments you've dropped in here on your new functions.